src/libs/Notification/PushNotification/index.native.ts

```diff
@@ -97,14 +97,13 @@ function init() {
     Airship.addListener(EventType.NotificationOptInStatus, refreshNotificationOptInStatus);
 
     ForegroundNotifications.configureForegroundNotifications();
-}
+};
 
 /**
  * Register this device for push notifications for the given notificationID.
- *
- * @param {String|Number} notificationID
  */
-function register(notificationID) {
+const register: Register = (notificationID) => {
+    // @ts-expect-error FIXME: This condition will never satisfy because Airship.contact.getNamedUserId() returns a promise.
```

Comment:
I will contact an Expensify engineer that could help me with this issue.

Reply:
@arosiclair Could you help me with this one since you changed this line [here](https://github.com/Expensify/App/commit/fc038c08f9b163ae03be52e5d9d90d2a3c5cf7fb)? The problem here is that `Airship.contact.getNamedUserId()` returns a `Promise`, so this condition will never satisfy the way it is today. What's your suggestion to fix this problem?

Reply:
Good catch looks like this check has been quietly broken. We can add an `await` here to get the proper value. I assume it's not a problem if this function is made `async`?

Reply:
I think we won't be able to use `async / await` due to restrictions in the project's guidelines, so I would have to do something like this. WDYT @arosiclair?

```ts
const register: Register = (notificationID) => {
    Airship.contact
        .getNamedUserId()
        .then((userID) => {
            if (userID === notificationID.toString()) {
                // No need to register again for this notificationID.
                return;
            }

            // Get permissions to display push notifications (prompts user on iOS, but not Android)
            Airship.push.enableUserNotifications().then((isEnabled) => {
                if (isEnabled) {
                    return;
                }

                Log.info('[PushNotification] User has disabled visible push notifications for this app.');
            });

            // Register this device as a named user in AirshipAPI.
            // Regardless of the user's opt-in status, we still want to receive silent push notifications.
            Log.info(`[PushNotification] Subscribing to notifications`);
            Airship.contact.identify(notificationID.toString());

            // Refresh notification opt-in status NVP for the new user.
            refreshNotificationOptInStatus();
        })
        .catch((error) => {
            Log.warn('[PushNotification] Failed to register for push notifications! Reason: ', error);
        });
};
```

Reply:
Yeah bummer about no async/await. That looks good to me 👍